src/twisted/web/test/test_http2.py

```diff
@@ -2918,3 +2924,34 @@ def test_timeOutClientThatSendsOnlyInvalidFrames(self):
         # Invalid frames don't reset any timeouts, so the above has
         # forcibly disconnected us via abortConnection.
         self.assertTrue(transport.disconnected)
+
+
+class EndToEndTests(unittest.TestCase):
+    """
+    Tests that run real TCP-based communication with a HTTP/2 server.
+    """
+
+    if skipH2 or not requireModule("OpenSSL"):
+        skip = skipH2
+
+    async def test_realRequest(self):
```

Comment:
```suggestion
    async def test_realRequest(self) -> None:
```

Reply:
let's not add more untyped code if we can help it

Reply:
Neat, I assume this coroutine is running because it's covered, when did we add that to trial?

Reply:
Not sure but I switched to this style of async test in Tahoe a while ago.